### PR TITLE
ray tune related stuff work properly

### PR DESCRIPTION
**Pull request by suzannejin**

### CODE REVIEW

**Comment by suzannejin** [comment]
@JoseEspinosa may i ask why there are two pipeline CI for test data? What are the differences?

File: bin/tests/test_model/titanic_model.py [hunk 1]
```diff
@@ -23,10 +23,13 @@ def forward(self, pclass: torch.tensor, sex: torch.tensor, age: torch.tesnor, si
     def compute_loss(self, loss_fn, output, survived):
         return loss_fn(output, survived)
     
-    def batch(self, x: dict, y: dict, loss_fn: Callable, optimizer: Callable):
-
+    def batch(self, x: dict, y: dict, loss_fn: Callable, optimizer: Callable = None):
         output = self.forward(**x)
         loss = self.compute_loss(loss_fn, output, **y)
+
+        if optimizer is None:
```
Comment (luisas) [inline bin/tests/test_model/titanic_model.py:30]: do we want to allow this? 

When do we have optimizer None? maybe we should instead have a default?

File: modules/local/torch_tune.nf [hunk 2]
```diff
@@ -10,7 +10,14 @@ process TORCH_TUNE {
 
     output:
     // TODO get the best model as well once implemented in python
-    tuple val(original_csv), path("best_config.json"), path(data_csv), path(parsed_json), emit: tune_specs
+    tuple val(original_csv), 
+          path("best_config.json"), 
+          path("best_model.pt"),
+          path("best_optimizer.pt"),
```
Comment (luisas) [inline modules/local/torch_tune.nf:16]: why are we also saving the optimizer?

File: bin/src/learner/raytune_parser.py
Comment (luisas) [inline bin/src/learner/raytune_parser.py:1]: Why not having these methods as methods of the TuneModel?

Comment (suzannejin) [reply bin/src/learner/raytune_parser.py:1]: i discussed at some point with mathys last friday, and he said that it is better and cleaner to have one class to do one main thing, and probably we should have a TuneModel and a TuneParser

File: bin/tests/test_model/titanic_model.py (hunk 1 above)
Comment (suzannejin) [reply bin/tests/test_model/titanic_model.py:30]: okay, i agree, i will change this

File: modules/local/torch_tune.nf (hunk 2 above)
Comment (luisas) [reply modules/local/torch_tune.nf:16]: ok!

File: bin/src/learner/raytune_parser.py
Comment (luisas) [reply bin/src/learner/raytune_parser.py:1]: ok

**Review by luisas — APPROVED** [review]

**Comment by JoseEspinosa** [comment]
> @JoseEspinosa may i ask why there are two pipeline CI for test data? What are the differences?

Since one is run with the last nextflow version and the other with `23.04.0`, to make sure it works with the last version but also with the version minimal supported by nf-core

**Pull request merged**